src/sdi-snap.c

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (C) 2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "sdi-snap.h"
+
+enum { PROP_NAME = 1, PROP_DIALOG, PROP_LAST };
+
+struct _SdiSnap {
+  GObject parent_instance;
+
+  gchar *name;
```

Comment:
Would be nice to have some comments on these fields.

Reply:
Good point. Done.

Reply:
Added. Check them to see if they are fine, please.